src/js/components/covid19/amountsVisualization/AmountsVisualization.jsx

```diff
@@ -0,0 +1,221 @@
+            });
+        }
+        else if (browser.includes('Firefox') || browser.includes('Safari')) {
+            // vertical offsets from trial/error. Not sure which element's height requires this?
+            const verticalOffset = window.innerWidth >= 1600
+                ? 0 + stickyHeaderHeight
+                : 29.5 + stickyHeaderHeight;
+            setMouseValue({
+                x: e.clientX - document.getElementById('amounts-viz_id').getBoundingClientRect().left,
+                y: e.clientY - document.getElementById('amounts-viz_id').getBoundingClientRect().top - verticalOffset
+            });
+        }
+        else {
+            setMouseValue({
+                x: e.offsetX || e.clientX,
+                y: e.offsetY || e.clientY
+            });
+        }
+    }, 100);
+
+    useEffect(() => {
+        document.getElementById('amounts-viz_id').addEventListener('mousemove', setMouseData);
+        return () => document.getElementById('amounts-viz_id').removeEventListener('mousemove', setMouseData);
+    }, []);
+
+    const tooltipForDefCodes = () => (showTooltip === `_totalBudgetAuthority` ?
+        `${calculatePercentage(overviewDataByDefCode[showTooltip], overviewData.all._totalBudgetAuthority, null, 2, { absoluteMin: 'Less than 0.01%' })} of Total Budgetary Resources for COVID-19 spending` :
+        `${calculatePercentage(overviewDataByDefCode[showTooltip], overviewDataByDefCode._totalBudgetAuthority, null, 2, { absoluteMin: 'Less than 0.01%' })} of Total Budgetary Resources for the ${tooltipShortName[publicLawFilter]}`);
+
+    const tooltipData = () => {
+        const percentage = publicLawFilter !== 'all' ? tooltipForDefCodes() :
+            `${calculatePercentage(overviewDataByDefCode[showTooltip], overviewDataByDefCode._totalBudgetAuthority, null, 2)} of Total Budgetary Resources`;
+        return {
+            tooltipPosition: 'bottom',
+            styles: {
+                position: 'absolute',
+                transform: `translate(${mouseValue.x - (defaultTooltipWidth / 2)}px,${mouseValue.y + 10}px)`
+            },
+            tooltipComponent: <PaginatedTooltipContainer
+                data={[{
+                    title: tooltipMapping[showTooltip].title,
+                    sections: [
+                        {
+                            paragraphs: [
+                                `${formatMoney(overviewDataByDefCode[showTooltip])}`,
+                                percentage,
+                                tooltipMapping[showTooltip].paragraph
+                            ]
+                        }
+                    ]
+                }]
+                }
+                tooltipElement={<Tooltip />} />
+        };
+    };
+
+    const displayTooltip = (e) => {
+        setShowTooltip(e.target.getAttribute('data-id'));
+    };
+
+    const hideTooltip = () => {
+        setShowTooltip('');
+    };
+
+    const preposition = publicLawFilter !== 'all' ? 'through the ' : '';
```

Comment:
wow nicely done sir. this is the most impressive variable name of your career. Gold star.